sdk/python/feast/registry.py

```diff
@@ -35,26 +36,33 @@
 
 REGISTRY_SCHEMA_VERSION = "1"
 
+warnings.simplefilter("once", DeprecationWarning)
+
 
 class Registry:
     """
     Registry: A registry allows for the management and persistence of feature definitions and related metadata.
     """
 
-    cached_registry_proto: Optional[RegistryProto] = None
-    cached_registry_proto_created: Optional[datetime] = None
-    cached_registry_proto_ttl: timedelta
-    cache_being_updated: bool = False
+    cached_registry_proto: RegistryProto
 
-    def __init__(self, registry_path: str, repo_path: Path, cache_ttl: timedelta):
+    def __init__(
+        self,
+        registry_path: str,
+        repo_path: Path,
+        cache_ttl: timedelta = None,
+        from_feast=False,
```

Comment:
nit, I think we want a better param than `from_feast`. Maybe `instantiated_internally`? Open to other ideas.

Reply:
That's a good point. `instantiated_internally` is much more clear; I can't think of anything better, so I'll go with that.